builtin/providers/aws/resource_aws_iot_policy_attachment.go

```diff
@@ -0,0 +1,151 @@
+package aws
+
+import (
+	"log"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/iot"
+	"github.com/hashicorp/terraform/helper/schema"
+)
+
+func resourceAwsIotPolicyAttachment() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceAwsIotPolicyAttachmentCreate,
+		Read:   resourceAwsIotPolicyAttachmentRead,
+		Update: resourceAwsIotPolicyAttachmentUpdate,
+		Delete: resourceAwsIotPolicyAttachmentDelete,
+		Schema: map[string]*schema.Schema{
+			"name": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"principals": &schema.Schema{
+				Type:     schema.TypeSet,
+				Required: true,
+				Elem: &schema.Schema{
+					Type: schema.TypeString,
+				},
+			},
+			"policy": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+			},
+		},
+	}
+}
+
+func resourceAwsIotPolicyAttachmentCreate(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).iotconn
+
+	for _, p := range d.Get("principals").(*schema.Set).List() {
+		_, err := conn.AttachPrincipalPolicy(&iot.AttachPrincipalPolicyInput{
```

Comment:
Is there a potential eventual consistency issue here? Do we need to wait and check the attachment has happened before we move on to the next?